### PR TITLE
Legacy Android Script Deprecation

### PR DESCRIPTION
**Pull request by spham-amzn**
## What does this PR do?
This PR deprecates the legacy android scripts that relied on gradle 7.0 to use the updated android generation scripts from https://github.com/o3de/o3de/pull/17078.

* Updates to support deprecation of legacy android generation scripts
* Update gradle to version 8.7 for the new android project generation workflow
* Update JDK to jdk17 for the new android project generation workflow to support the latest android_sdk
* Add new 'NONE' option to 'ASSET_MODE' to support generating APK packages without assets (needed for gradle AR, but still a valid use case)
* Update the gradle_windows.cmd to use the new android project generation scripts in the o3de command line tools

## How was this PR tested?

```
python\python.cmd scripts\build\ci_build.py -p Android -t gradle
```

Addresses https://github.com/o3de/o3de/issues/18013

### CODE REVIEW

**Review by nick-l-o3de — APPROVED** [review]
Is this supposed to work on linux too?

**Comment by spham-amzn** [comment]
> Is this supposed to work on linux too?

This deprecation focuses mostly on the AR portion. The new android generation scripts and workflow is suppose to work on both windows and linux.

**Review by lemonade-dm — APPROVED** [review]

**Review by amzn-changml — APPROVED** [review]
Looks reasonable. Will need a node update before merging/AR

**Comment by spham-amzn** [comment]
> > Is this supposed to work on linux too?
> 
> This deprecation focuses mostly on the AR portion. The new android generation scripts and workflow is suppose to work on both windows and linux.

Correction, it almost works :-) 

https://github.com/o3de/o3de/pull/18123

**Review by amzn-changml — COMMENTED** [review]
Some suggested fixes for the powershell script as it was being deployed.

File: scripts/build/build_node/Platform/Windows/install-windows-android.ps1 [hunk 1]
```diff
@@ -5,9 +5,20 @@ For complete copyright and license terms please see the LICENSE at the root of t
...
+#Android SDK Command line tools needs a custom installer due the latest command like version not being available in the Chocolatey repository
+$androidSdkPackageName = 'androidsdk'
+$androidSdkVersion = '11076708_latest'
+$androidSdkUrl = "https://dl.google.com/android/repository/commandlinetools-win-$androidSdkVersion.zip"
+$androidSdkChecksum = '4d6931209eebb1bfb7c7e8b240a6a3cb3ab24479ea294f3539429574b1eec862'
+$androidSdkInstallDir = "C:\AndroidSdk\cmdline-tools"
+Install-ChocolateyZipPackage $androidSdkPackageName $androidSdkUrl $androidSdkInstallDir -Checksum $androidSdkChecksum -ChecksumType 'sha256'
+[Environment]::SetEnvironmentVariable("ANDROID_SDK_ROOT", "C:\AndroidSdk", [EnvironmentVariableTarget]::Machine)
+# Rename the Android SDK folder 'cmdline-tools' to match what the command line tools looks like normally in a fulll AndroidSDK install
+Rename-Item -Path "C:\AndroidSdk\cmdline-tools\cmdline-tools" "C:\AndroidSdk\cmdline-tools\latest"
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply scripts/build/build_node/Platform/Windows/install-windows-android.ps1:21]: Yeah,m its baffling why they hard coded the paths like like.

File: scripts/build/Platform/Android/pipeline.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "ENV": {
         "GRADLE_HOME": "C:/Gradle/gradle-7.0",
-        "NODE_LABEL": "windows-latest",
+        "NODE_LABEL": "windows-2024-07",
         "LY_3RDPARTY_PATH": "D:\\workspace\\3rdParty",
         "LY_NDK_DIR": "C:/Android/android-sdk/ndk/21.4.7075529",
```
Comment (amzn-changml) [inline scripts/build/Platform/Android/pipeline.json:6]: Are we using the older `Android` directory or the new one from `AndroidSdk`? If the latter:
```suggestion
        "LY_NDK_DIR": "C:/AndroidSdk/ndk/21.4.7075529",
```

File: scripts/build/build_node/Platform/Windows/install-windows-android.ps1 [hunk 3]
```diff
@@ -16,15 +28,15 @@ $build_tools = '"build-tools;30.0.2" "build-tools;34.0.0" "tools"'
 $ndk = '"ndk;21.4.7075529" "ndk;23.1.7779620" "ndk;25.1.8937393" "ndk;25.2.9519653"'
 
 Write-Host "Installing Android SDK packages"
-$sdkmanager = "C:\Android\android-sdk\tools\bin\sdkmanager.bat"
+$sdkmanager = "$androidSdkLatestDir\bin\sdkmanager.bat"
 Start-Process -FilePath $sdkmanager -ArgumentList $android_packages -NoNewWindow -Wait
 Start-Process -FilePath $sdkmanager -ArgumentList $googleplay_packages -NoNewWindow -Wait
 Start-Process -FilePath $sdkmanager -ArgumentList $build_tools -NoNewWindow -Wait
 Start-Process -FilePath $sdkmanager -ArgumentList $ndk -NoNewWindow -Wait
```
Comment (amzn-changml) [inline scripts/build/build_node/Platform/Windows/install-windows-android.ps1:35]:
```suggestion
Start-Process -FilePath $sdkmanager -ArgumentList "--licenses" -NoNewWindow -Wait # Accept all licenses manually
Start-Process -FilePath $sdkmanager -ArgumentList $android_packages -NoNewWindow -Wait
Start-Process -FilePath $sdkmanager -ArgumentList $googleplay_packages -NoNewWindow -Wait
Start-Process -FilePath $sdkmanager -ArgumentList $build_tools -NoNewWindow -Wait
Start-Process -FilePath $sdkmanager -ArgumentList $ndk -NoNewWindow -Wait
```

**Pull request merged**